QUARKUS-5613.md

```diff
@@ -0,0 +1,85 @@
+## Links to the resources
+- JIRA issue: https://issues.redhat.com/browse/QUARKUS-5613
+- GH PRs related to the Jira issue: https://github.com/quarkusio/quarkus/pull/43283 https://github.com/quarkusio/quarkus/issues/42117 
+- Guides modified and related:
+  https://quarkus.io/guides/security-openid-connect-client-reference
+  https://quarkus.io/guides/security-oidc-code-flow-authentication
+  https://quarkus.io/guides/security-openid-connect-client-registration
+- TP/TD PRs:
+
+## Scope of the testing
+
+Our main testing scope is to ensure that the new response filter mechanism works as intended for various OIDC endpoints, ensuring that is possible intercept and handle responses
+from various OIDC endpoints.
+
+This complements the already available OidcRequestFilter functionality.
+
+Integration tests will be on the next module in our Test suite:    
+ security/keycloak-oidc-client-reactive-extended (https://github.com/quarkus-qe/quarkus-test-suite/tree/main/security/keycloak-oidc-client-reactive-extended).
+And the scope will be covered on areas related to:
+   - Security
+   - Logging /Tracing - Enabling TRACE level logs in properties will capture relevant info for our tests
+
+The test coverage will be verified for JVM, native mode, and Openshift.
+
+The testing include both OIDC Client (quarkus-oidc) and OIDC Client (quarkus-oidc-client) extensions.
+
+### What to Test - Scenarios
+
+#### Token Endpoint Response Filtering
+ - Verify that a custom OIDC response filter (TokenEndpointResponseFilter) is triggered immediately after a successful token refresh from Keycloak.
+ - Validation
+    * Ensure that filter logs messages contains "token have been refreshed".
+    * Verify 'OidcResponseContext' exposes the correct property grant_type=refresh_token
+    * Assert that the response includes "refresh_token"
+
+####  User Info Endpoint Response Filtering
+- Demonstrate that an OIDC response filter can intercept the User Info endpoint response, for instance verifying user claims.
+-  Validation
+    * Filter runs and logs claim details(sub,preferred_username)
+    * JSON body response includes the details.
+    * Logs show filter intercepting '/userinfo' response
+
+#### Client Registration Endpoint Response Filtering
+- Create a filter(OidcEndpoint.Type.CLIENT_REGISTRATION) intercepts responses to client registration calls.
+- Validation
+  * Registration data, for instance check filter logs newly registered clint_id or client_name
+  * Check if filter can detect a missing invalid property in the returned JSON
+  * Error handling, if Keycloak return a 401 the filter log propagate it correctly.
+
+All scenarios will be tested in both JVM and Native mode on baremetal and Openshift.
+
+
+## Getting familiar with the feature
+- Focus on exploratory testing of the Keycloak and OIDC testing modules
```

Comment:
That sounds very wrong. You fight for users, you are user champion! Do you expect Quarkus users to learn how to use this feature by reading Quarkus tests? Please focus on customer perspective.

Reply:
e.g. documentation

Reply:
You're right, I am removing that sentence that was for me exploring what we have about it...